packages/spring/src/index.ts

```diff
@@ -0,0 +1,319 @@
+  now: () => number;
+  /** A set of tasks that will run to completion, unless aborted */
+  tasks: Set<TaskEntry>;
+} = {
+  tick: (_: any) => (isServer ? () => {} : requestAnimationFrame(_)), // SSR-safe RAF function.
+  now: () => performance.now(), // Getter for now() using performance in browser and Date in server. Although both are available in node and browser.
+  tasks: new Set(),
+};
+
+// https://github.com/sveltejs/svelte/blob/main/packages/svelte/src/motion/utils.js
+
+function is_date(obj: any): obj is Date {
+  return Object.prototype.toString.call(obj) === "[object Date]";
+}
+
+// https://github.com/sveltejs/svelte/blob/main/packages/svelte/src/internal/client/loop.js
+
+function run_tasks(now: number) {
+  raf.tasks.forEach(task => {
+    if (!task.c(now)) {
+      raf.tasks.delete(task);
+      task.f();
+    }
+  });
+
+  if (raf.tasks.size !== 0) {
+    raf.tick(run_tasks);
+  }
+}
+
+/**
+ * Creates a new task that runs on each raf frame
+ * until it returns a falsy value or is aborted
+ */
+function loop(callback: TaskCallback): Task {
+  let task: TaskEntry;
+
+  if (raf.tasks.size === 0) {
+    raf.tick(run_tasks);
+  }
+
+  return {
+    promise: new Promise((fulfill: any) => {
+      raf.tasks.add((task = { c: callback, f: fulfill }));
+    }),
+    abort() {
+      raf.tasks.delete(task);
+    },
+  };
+}
+
+// ===========================================================================
+// createSpring hook
+// ===========================================================================
+
+import { Accessor, createEffect, createSignal, untrack } from "solid-js";
+
+export type SpringOptions = {
+  /**
+   * Stiffness of the spring. Higher values will create more sudden movement.
+   * @default 0.15
+   */
+  stiffness?: number;
+  /**
+   * Strength of opposing force. If set to 0, spring will oscillate indefinitely.
+   * @default 0.8
+   */
+  damping?: number;
+  /**
+   * Precision is the threshold relative to the target value at which the
+   * animation will stop based on the current value.
+   *
+   * From 0, if the target value is 500, and the precision is 500, it will stop
+   * the animation instantly (no animation, similar to `hard: true`).
+   *
+   * From 0, if the target value is 500, and the precision is 0.01, it will stop the
+   * animation when the current value reaches 499.99 or 500.01 (longer animation).
+   *
+   * @default 0.01
+   */
+  precision?: number;
+};
+
+export type SpringTargetPrimitive = number | Date;
+export type SpringTarget =
+  | SpringTargetPrimitive
+  | { [key: string]: SpringTargetPrimitive | SpringTarget }
+  | SpringTargetPrimitive[]
+  | SpringTarget[];
+
+/**
+ * "Widen" Utility Type so that number types are not converted to
+ * literal types when passed to `createSpring`.
+ *
+ * e.g. createSpring(0) returns `0`, not `number`.
+ */
+export type WidenSpringTarget<T> = T extends number ? number : T extends Date ? Date : T;
```

Comment:
I understand `T extends number`, but what is `T extends Date` for?

Reply:
Hmm. Come to think of it, you're right.

`number` sometimes becomes 0 or 1 (depending on what was passed) so this is needed for that.

`Date` doesn't really get inferred as anything else but Date, so no need for that `T extends Date`

Can just do:
`export type WidenSpringTarget<T> = T extends number ? number : T`